ghost/core/core/server/data/migrations/versions/5.55/2023-07-10-05-16-55-add-built-in-collection-posts.js

```diff
@@ -0,0 +1,99 @@
+const logging = require('@tryghost/logging');
+const {default: ObjectID} = require('bson-objectid');
+const {createTransactionalMigration} = require('../../utils');
+
+const insertPostCollections = async (knex, collectionId, postIds) => {
+    logging.warn(`Batch inserting ${postIds.length} collection posts for collection ${collectionId}`);
+
+    const collectionPosts = postIds.map((postId) => {
+        return {
+            id: (new ObjectID()).toHexString(),
+            collection_id: collectionId,
+            post_id: postId,
+            sort_order: 0
+        };
+    });
+
+    await knex.batchInsert('collections_posts', collectionPosts, 1000);
+};
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        logging.info('Populating built-in collections');
+
+        const existingLatestCollection = await knex('collections')
+            .where({
+                slug: 'latest'
+            })
+            .first();
+
+        if (!existingLatestCollection) {
+            logging.warn('Latest collection does not exists, skipping');
+        } else {
+            const latestPostsRows = await knex('posts')
+                .select('id')
+                .where({
+                    type: 'post'
```

Comment:
Does this need a filter for `status: 'published'`?

Reply:
nope. we would store all the posts on the db layer and only filter by published on the application layer. the fixture for "latest" collection has no filter at the moment at all.

Reply:
Why do we do that? Is it going to make our pagination more difficult?

Reply:
It does, but keeps the separation of logic clean (we had a very brief chat about this [here](https://ghost.slack.com/archives/C05DK3SP801/p1688723868922009))